contrib/simple-client.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: LGPL-2.1+
+
+"""A simple fwupd frontend"""
+import sys
+import os
+import gi
+gi.require_version('Fwupd', '2.0')
+from gi.repository import Fwupd
+from gi.repository import GLib
+
+class Progress ():
+        def __init__ (self):
+                self.device = None
+                self.status = None
+                self.percent = 0
+                self.erase = 0
+
+        def device_changed (self, new_device):
+                if self.device != new_device:
+                        self.device = new_device
+                        print ("\nUpdating %s" % self.device)
+
+        def status_changed (self, percent, status):
+                if (self.status != status) or (self.percent != percent):
+                        for i in range (0, self.erase):
+                                sys.stdout.write("\b \b")
+                        self.status = status
+                        self.percent = percent
+                        str = "["
+                        for i in range (0, 50):
+                                if i < percent/2:
+                                        str += '*'
+                                else:
+                                        str += ' '
+                        str += "] %d%% %s" % (percent, status)
+                        self.erase = len(str)
+                        sys.stdout.write(str)
+                        sys.stdout.flush()
+                        if 'idle' in status:
+                                sys.stdout.write ("\n")
+
+def parse_args():
+    import argparse
+    parser = argparse.ArgumentParser(description="Interact with fwupd daemon")
+    parser.add_argument("--cab", default="", help="CAB file to flash")
```

Comment:
isn't it odd to have this rather than putting the argv after `install` like `fwupdmgr`?